src/main/2.0/scala/com/holdenkarau/spark/testing/DataFrameSuiteBase.scala

```diff
@@ -158,21 +161,44 @@ trait DataFrameSuiteBaseLike extends SparkContextProvider
   }
 
   /**
-   * Zip RDD's with precise indexes. This is used so we can join two DataFrame's
-   * Rows together regardless of if the source is different but still compare
-   * based on the order.
-   */
+    * Zip RDD's with precise indexes. This is used so we can join two DataFrame's
+    * Rows together regardless of if the source is different but still compare
+    * based on the order.
+    */
   private[testing] def zipWithIndex[U](rdd: RDD[U]) = {
-    rdd.zipWithIndex().map{ case (row, idx) => (idx, row) }
+    rdd.zipWithIndex().map { case (row, idx) => (idx, row) }
   }
 
   def approxEquals(r1: Row, r2: Row, tol: Double): Boolean = {
     DataFrameSuiteBase.approxEquals(r1, r2, tol)
   }
+
+  def schemaEquals(expected: DataFrame, result: DataFrame): Unit = {
+    val expectedSchema = expected.schema
+    val resultSchema = result.schema
+    val errorString = schemaErrorMessage(expectedSchema, resultSchema)
```

Comment:
I see two options one is identify equality and when that fails use `assert(errorString, expectedSchema, resultSchema)` apparent problem with this approach is equality is checked in two places now. 
Other approach is to throw Custom Exceptions like 
`case class DataFrameSchemaMismatch(smth: String) extends Exception(smth)`
instead of relying on `assert`.
Example [here](https://github.com/MrPowers/spark-fast-tests/blob/master/src/main/scala/com/github/mrpowers/spark/fast/tests/DataFrameComparer.scala#L46)